### PR TITLE
[Visual] Visual tidy

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###
- Tidy up Material Renderers 
- remove flag
- Change the namespace for material to all be Xamarin.Forms.Material.<Platform> which is more consistent with additional things like Maps

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- Android
- UWP

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by mattleibow** [comment]
Looks good so far.

File: Xamarin.Forms.Core/Xamarin.Forms.Core.csproj [hunk 1]
```diff
@@ -22,17 +22,8 @@
...
-		Name="XFCorePostProcessor"
-		Inputs="$(IntermediateOutputPath)$(TargetFileName)"
-		Outputs="$(IntermediateOutputPath)XFCorePostProcessor.stamp">
-	    <Touch
-		    Files="$(IntermediateOutputPath)XFCorePostProcessor.stamp"
-		    AlwaysCreate="True" />
-	    <FixXFCoreAssembly
-		    Assembly="$(IntermediateOutputPath)$(TargetFileName)"
-		    ReferencePath="@(ReferencePath)" />
+	<Target Condition="$(DesignTimeBuild) != true AND $(BuildingProject) == true" AfterTargets="AfterCompile" Name="XFCorePostProcessor" Inputs="$(IntermediateOutputPath)$(TargetFileName)" Outputs="$(IntermediateOutputPath)XFCorePostProcessor.stamp">
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Xamarin.Forms.Core.csproj]: I like the way they are. don't change formatting if you do not change the actual content

File: Xamarin.Forms.Material.iOS/MaterialActivityIndicatorRenderer.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ public class MaterialActivityIndicatorRenderer : ViewRenderer<ActivityIndicator,
 
 		public MaterialActivityIndicatorRenderer()
 		{
-			VisualElement.VerifyVisualFlagEnabled();
+
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialActivityIndicatorRenderer.cs]: if you actually plan to tidy up, avoid white lines

File: Xamarin.Forms.Material.iOS/MaterialButtonRenderer.cs [hunk 3]
```diff
@@ -26,7 +26,7 @@ public class MaterialButtonRenderer : ViewRenderer<Button, MButton>, IImageVisua
 
 		public MaterialButtonRenderer()
 		{
-			VisualElement.VerifyVisualFlagEnabled();
+
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialButtonRenderer.cs]: if you actually plan to tidy up, avoid white lines

File: Xamarin.Forms.Material.iOS/MaterialDatePickerRenderer.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+			ApplyTheme();
+		}
+
+		protected internal override void UpdateFont()
+		{
+			base.UpdateFont();
+			Control?.ApplyTypographyScheme(Element);
+		}
+		
+
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialDatePickerRenderer.cs]: if you actually plan to tidy up, avoid white lines

File: Xamarin.Forms.Material.iOS/MaterialDatePickerRenderer.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
+﻿using System.ComponentModel;
+using UIKit;
+
+namespace Xamarin.Forms.Platform.iOS.Material
+{
+	public class MaterialDatePickerRenderer : DatePickerRendererBase<MaterialTextField>, IMaterialEntryRenderer
+	{
+		public MaterialDatePickerRenderer()
+		{
+
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialDatePickerRenderer.cs]: if you actually plan to tidy up, avoid white lines

File: Xamarin.Forms.Material.iOS/MaterialDatePickerRenderer.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	{
+		public MaterialDatePickerRenderer()
+		{
+			
+		}
+
+		protected override MaterialTextField CreateNativeControl()
+		{
+			var field = new NoCaretMaterialTextField(this, Element);
+			return field;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialDatePickerRenderer.cs]: what's the use of the local variable here ?

File: Xamarin.Forms.Material.iOS/MaterialDatePickerRenderer.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+			Control?.ApplyTypographyScheme(Element);
+		}
+		
+
+		protected internal override void UpdateTextColor()
+		{
+			Control?.UpdateTextColor(this);
+		}
+
+
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialDatePickerRenderer.cs]: if you actually plan to tidy up, avoid white lines

File: Xamarin.Forms.Material.iOS/MaterialDatePickerRenderer.cs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+		protected override void OnElementChanged(ElementChangedEventArgs<DatePicker> e)
+		{
+			base.OnElementChanged(e);
+			UpdatePlaceholder();
+		}
+
+		string IMaterialEntryRenderer.Placeholder => string.Empty;
+		Color IMaterialEntryRenderer.PlaceholderColor => Color.Default;
+
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialDatePickerRenderer.cs]: if you actually plan to tidy up, avoid white lines

File: Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+		{			
+			return new MaterialMultilineTextField(this, Element);
+		}
+
+		protected override void SetBackgroundColor(Color color)
+		{
+			ApplyTheme();			
+		}
+
+
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs]: if you actually plan to tidy up, avoid white lines

File: Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+		protected internal override void UpdateTextColor()
+		{
+			Control?.UpdateTextColor(this);
+		}
+
+
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs]: if you actually plan to tidy up, avoid white lines

File: Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+using MaterialComponents;
+using System.Threading.Tasks;
+using Foundation;
+using System;
+
+namespace Xamarin.Forms.Platform.iOS.Material
+{
+	public class MaterialEditorRenderer : EditorRendererBase<MaterialMultilineTextField>, IMaterialEntryRenderer
+	{
+		bool _hackHasRan = false;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs]: we'll regret this one day

File: Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+			InitialPlaceholderSetupHack();
+		}
+
+		protected internal override void UpdateAutoSizeOption()
+		{
+			base.UpdateAutoSizeOption();
+			Control.ExpandsOnOverflow = Element.AutoSize == EditorAutoSizeOption.TextChanges;
+		}
+
+
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs]: if you actually plan to tidy up, avoid white lines

File: Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+				_hackHasRan = true;
+				UpdateText();
+				TextView.ResignFirstResponder();
+			});
+		}
+
+
+		// Placeholder is currently broken upstream and doesn't animate to the correct scale
+		string IMaterialEntryRenderer.Placeholder => Element?.Placeholder;
+		// string IMaterialEntryRenderer.Placeholder => String.Empty;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs]: why the commented code ?

File: Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs]: why the white line ?

File: Xamarin.Forms.Material.iOS/MaterialEntryRenderer.cs [hunk 15]
```diff
@@ -1,145 +1,54 @@
...
-		}
-
-		public override CGSize SizeThatFits(CGSize size)
-		{
-			var result =  base.SizeThatFits(size);
-			if (nfloat.IsInfinity(result.Width))
-				result = Control.SystemLayoutSizeFittingSize(result, (float)UILayoutPriority.FittingSizeLevel, (float)UILayoutPriority.DefaultHigh);
-
-			return result;
+
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.iOS/MaterialEntryRenderer.cs]: if you actually plan to tidy up, avoid white lines

File: Xamarin.Forms.Core/Xamarin.Forms.Core.csproj (hunk 1 above)
Comment (PureWeen) [reply Xamarin.Forms.Core/Xamarin.Forms.Core.csproj]: I don't think we're going to get around this change. The IDE is making this change so it's going to cause a lot of headaches for everyone to selectively not commit this

File: Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs (hunk 11 above)
Comment (PureWeen) [reply Xamarin.Forms.Material.iOS/MaterialEditorRenderer.cs]: Once we update to latest iOS Material Components we can remove this code

**Comment by PureWeen** [comment]
@StephaneDelcroix  apologies on the state first review through. The base branch got off a few moments before you started so there were a lot of extraneous files in the commits on this one. It should be more inline with actual changes now

File: Xamarin.Forms.Material.iOS/MaterialMultiLineTextField.cs [hunk 16]
```diff
@@ -10,19 +10,17 @@ namespace Xamarin.Forms.Platform.iOS.Material
 {
 	public class MaterialMultilineTextField : MMultilineTextField, IMaterialTextField
 	{
+
+		public MaterialMultilineTextField(IMaterialEntryRenderer element, IFontElement fontElement) => MaterialTextManager.Init(element, this, fontElement);
+
+
```
Comment (mattleibow) [inline Xamarin.Forms.Material.iOS/MaterialMultiLineTextField.cs]: Lots of whitespace here ;)

Comment (PureWeen) [reply Xamarin.Forms.Material.iOS/MaterialMultiLineTextField.cs]: Should be fixed now

**Pull request merged**